### PR TITLE
[CHI-70] Add breadcrumb component.

### PR DESCRIPTION
**Pull request by davidojedacl**
@mattnickles @jllr I have a doubt related base font size. If i don't use -text and I put it in the breadcrumb class the base font size then the text modifiers don't work

### CODE REVIEW

File: src/chi/components/breadcrumb/breadcrumb.scss [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        + .a-breadcrumb__item::before {
+          content: url('./icon-chevron-right.svg');
+          height: 1rem;
+          position: relative;
+          width: 1rem;
+          top: 0.1rem;
+        }
+      }
+    }
+    > ol {
```
Comment (mattnickles) [inline src/chi/components/breadcrumb/breadcrumb.scss]: Need to zero out the padding and margin on the `ol`

File: src/website/views/components/breadcrumb.pug [hunk 2]
```diff
@@ -1,4 +1,158 @@
...
+
+h3 Examples
+p.-text
+  | To render a breadcrumb, use the class <code>a-breadcrumb</code>.
+  | A contextual class (e.g. <code>-separatorIcon</code>) is required to provide alternative separator styling.
+
+h4 Base
+.example.-mb3
+  .-p1
+    nav.a-breadcrumb.-text
```
Comment (mattnickles) [inline src/website/views/components/breadcrumb.pug]: Let's add `aria-label="Breadcrumb"` to each `<nav>` element as defined [here](https://www.w3.org/TR/wai-aria-practices/examples/breadcrumb/index.html) in w3 best practices. We'll also need `aria-current="page"` on the anchor of the active page, in this case, Child 2.

File: src/website/views/components/breadcrumb.pug [hunk 3]
```diff
@@ -1,4 +1,158 @@
...
+p.-text Breadcrumb indicate the current page’s location within a navigational hierarchy.
+
+h3 Examples
+p.-text
+  | To render a breadcrumb, use the class <code>a-breadcrumb</code>.
+  | A contextual class (e.g. <code>-separatorIcon</code>) is required to provide alternative separator styling.
+
+h4 Base
+.example.-mb3
+  .-p1
```
Comment (mattnickles) [inline src/website/views/components/breadcrumb.pug]: Within each example, change `-p1` to `-p3` to give the sections more padding

File: src/chi/components/breadcrumb/breadcrumb.scss [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  .a-breadcrumb__item {
+    + .a-breadcrumb__item {
+      padding-left: .25rem;
+
+      &::before {
+        display: inline-block;
+        padding-right: .25rem;
+        color: set-color(gray, 50);
+        content: quote("/");
```
Comment (mattnickles) [inline src/chi/components/breadcrumb/breadcrumb.scss]: Add `font-weight: normal;` to the / before element.

File: src/chi/components/breadcrumb/breadcrumb.scss [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      }
+    }
+
+    + .a-breadcrumb__item:hover::before {
+      text-decoration: none;
+    }
+
+    &.-active {
+      color: set-color(gray, 50);
```
Comment (mattnickles) [inline src/chi/components/breadcrumb/breadcrumb.scss]: Active color should be the base text color

Comment (mattnickles) [inline src/chi/components/breadcrumb/breadcrumb.scss]: Also the active item should be set to `font-weight: normal;`

File: src/chi/components/breadcrumb/breadcrumb.scss [hunk 6]
```diff
@@ -0,0 +1,44 @@
+@import 'variables';
+@import 'mixins';
+
+.chi {
+  .a-breadcrumb {
+    &.-separatorIcon {
+      .a-breadcrumb__item {
+        + .a-breadcrumb__item::before {
```
Comment (mattnickles) [inline src/chi/components/breadcrumb/breadcrumb.scss]: &::before

File: src/website/views/components/breadcrumb.pug [hunk 7]
```diff
@@ -1,4 +1,158 @@
...
+          <li class="a-breadcrumb__item">
+            <a href="#">Child 1</a>
+          </li>
+          <li class="a-breadcrumb__item -active">
+            Child 2
+          </li>
+      </ol>
+    </nav>
+
+h4 Alternative Separator
```
Comment (mattnickles) [inline src/website/views/components/breadcrumb.pug]: Let's change the alternative modifier class from `-separatorIcon` to just `-alt`

File: src/website/views/components/breadcrumb.pug [hunk 8]
```diff
@@ -1,4 +1,158 @@
...
+    nav.a-breadcrumb.-text.-separatorIcon
+      ol
+        li.a-breadcrumb__item
+          a(href='#') Parent
+        li.a-breadcrumb__item
+          a(href='#') Child 1
+        li.a-breadcrumb__item.-active
+          | Child 2
+  :code(lang="html")
+    <nav class="a-breadcrumb -text -separatorIcon">
```
Comment (mattnickles) [inline src/website/views/components/breadcrumb.pug]: Place the `-alt` class before the `-text `modifier class in the examples

File: src/website/views/components/breadcrumb.pug [hunk 9]
```diff
@@ -1,4 +1,158 @@
...
+    nav.a-breadcrumb.-text
+      ol
+        li.a-breadcrumb__item
+          a(href='#') Parent
+        li.a-breadcrumb__item
+          a(href='#') Child 1
+        li.a-breadcrumb__item.-active
+          | Child 2
+  :code(lang="html")
+    <nav class="a-breadcrumb -text">
```
Comment (mattnickles) [inline src/website/views/components/breadcrumb.pug]: Let's render the base version at `font-size: .875rem;` so no `-text` class is needed on the base.

Comment (davidojedacl) [reply src/website/views/components/breadcrumb.pug]: If i put  font-size: .875rem; in the breadcrumb.scss base later the sizes modifers don't works

Comment (jllr) [reply src/website/views/components/breadcrumb.pug]: @davidojedacl set -text in <nav>, as you had in the first version, so we can move on.

**Pull request merged**